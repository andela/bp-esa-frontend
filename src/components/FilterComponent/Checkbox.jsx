/* eslint-disable jsx-a11y/label-has-associated-control, jsx-a11y/label-has-for */
import React from 'react';
import PropTypes from 'prop-types';

/**
 * Displays a custom checkbox
 */
const Checkbox = ({
  checkCondition, id, label, name, noLabel = false, ...rest
}) => {
  const Input = () => (
    <React.Fragment>
      <input
        type="checkbox"
        id={id}
        name={name}
        defaultChecked={checkCondition}
        {...rest}
      />
      <span className="checkmark" />
    </React.Fragment>
  );
  return (noLabel ? <Input /> : (
    <label className={`${id}-label`} htmlFor={id}>
      <Input />
      {label}
    </label>
  )
  );
};

Checkbox.defaultProps = {
  noLabel: false,
  label: '',
};
Checkbox.propTypes = {
  checkCondition: PropTypes.bool.isRequired,
  id: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  label: PropTypes.string,
  noLabel: PropTypes.bool,
};

export const generateCheckboxDuo = (name, filter, values, labels = []) => {
  const baseCheckbox = {
    name,
    noLabel: !labels.length,
  };
  return [{
    ...baseCheckbox,
    checkCondition: filter === values[0],
    value: values[0],
    label: labels[0] || '',
    id: `${name}-${values[0]}`,
  }, {
    ...baseCheckbox,
    checkCondition: filter === values[1],
    value: values[1],
    label: labels[1],
    id: `${name}-${values[1]}`,
  }];
};

export default Checkbox;
