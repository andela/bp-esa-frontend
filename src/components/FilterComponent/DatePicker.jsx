import React from 'react';
import PropTypes from 'prop-types';
import ReactDatePicker from 'react-datepicker';
import 'react-datepicker/dist/react-datepicker.css';
import moment from 'moment';

const DatePicker = (props) => {
  const {
    onChange, selection, dateSelected, ...rest
  } = props;
  return (
    <ReactDatePicker
      selected={dateSelected || undefined}
      onChange={date => onChange(date, selection)}
      dropdownMode="select"
      shouldCloseOnSelect={false}
      calendarClassName="custom-date-picker"
      dateFormat="MMMM DD, YYYY"
      maxDate={moment()}
      {...rest}
    />
  );
};


DatePicker.propTypes = {
  onChange: PropTypes.func.isRequired,
  selection: PropTypes.string.isRequired,
  dateSelected: PropTypes.oneOfType([PropTypes.string, PropTypes.object]).isRequired,
};

export default DatePicker;
