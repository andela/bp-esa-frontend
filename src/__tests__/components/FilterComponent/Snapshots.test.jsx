import React from 'react';
import FilterDropdown from '../../../components/FilterComponent/FilterDropdown';
import DatePicker from '../../../components/FilterComponent/DatePicker';
import FilterComponent, { filterInitialState } from '../../../components/FilterComponent';
import Checkbox from '../../../components/FilterComponent/Checkbox';

describe('The FilterDropdown component', () => {
  it('should not regress', () => {
    const filterDropdown = shallow(
      <FilterDropdown
        filters={filterInitialState}
        handleInputChange={() => {}}
      />,
    );
    expect(filterDropdown).toMatchSnapshot();
  });
});


describe('The Checkbox component', () => {
  it('should not regress(with label)', () => {
    const checkbox = shallow(
      <Checkbox
        checkCondition={false}
        id="test-checkbox"
        name="test"
        label="test checkbox"
      />,
    );
    expect(checkbox).toMatchSnapshot();
  });

  it('should not regress(without label)', () => {
    const checkbox = shallow(
      <Checkbox checkCondition={false} id="test-checkbox2" name="test2" noLabel />,
    );
    expect(checkbox).toMatchSnapshot();
  });
});


describe('The DatePicker component', () => {
  it('should not regress', () => {
    const datePicker = shallow(
      <DatePicker onChange={() => {}} selection="to" dateSelected="" />,
    );
    expect(datePicker).toMatchSnapshot();
  });
});

describe('The FilterComponent component', () => {
  it('should not regress', () => {
    const filterComponent = shallow(
      <FilterComponent filter={filterInitialState} />,
    );
    expect(filterComponent).toMatchSnapshot();
  });
});
