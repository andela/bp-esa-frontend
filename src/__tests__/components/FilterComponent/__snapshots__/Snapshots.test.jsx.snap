// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Checkbox component should not regress(with label) 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Checkbox
    checkCondition={false}
    id="test-checkbox"
    label="test checkbox"
    name="test"
    noLabel={false}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Input />,
        "test checkbox",
      ],
      "className": "test-checkbox-label",
      "htmlFor": "test-checkbox",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "test checkbox",
    ],
    "type": "label",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Input />,
          "test checkbox",
        ],
        "className": "test-checkbox-label",
        "htmlFor": "test-checkbox",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "test checkbox",
      ],
      "type": "label",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`The Checkbox component should not regress(without label) 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Checkbox
    checkCondition={false}
    id="test-checkbox2"
    label=""
    name="test2"
    noLabel={true}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {},
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {},
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`The DatePicker component should not regress 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <DatePicker
    dateSelected=""
    onChange={[Function]}
    selection="to"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "allowSameDay": false,
      "calendarClassName": "custom-date-picker",
      "dateFormat": "MMMM DD, YYYY",
      "dateFormatCalendar": "MMMM YYYY",
      "disabled": false,
      "disabledKeyboardNavigation": false,
      "dropdownMode": "select",
      "maxDate": "2019-09-13T06:49:17.309Z",
      "monthsShown": 1,
      "nextMonthButtonLabel": "Next month",
      "onBlur": [Function],
      "onChange": [Function],
      "onClickOutside": [Function],
      "onFocus": [Function],
      "onInputClick": [Function],
      "onInputError": [Function],
      "onKeyDown": [Function],
      "onMonthChange": [Function],
      "onSelect": [Function],
      "onYearChange": [Function],
      "preventOpenOnFocus": false,
      "previousMonthButtonLabel": "Previous Month",
      "readOnly": false,
      "renderDayContents": [Function],
      "selected": undefined,
      "shouldCloseOnSelect": false,
      "showTimeSelect": false,
      "timeCaption": "Time",
      "timeIntervals": 30,
      "withPortal": false,
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "allowSameDay": false,
        "calendarClassName": "custom-date-picker",
        "dateFormat": "MMMM DD, YYYY",
        "dateFormatCalendar": "MMMM YYYY",
        "disabled": false,
        "disabledKeyboardNavigation": false,
        "dropdownMode": "select",
        "maxDate": "2019-09-13T06:49:17.309Z",
        "monthsShown": 1,
        "nextMonthButtonLabel": "Next month",
        "onBlur": [Function],
        "onChange": [Function],
        "onClickOutside": [Function],
        "onFocus": [Function],
        "onInputClick": [Function],
        "onInputError": [Function],
        "onKeyDown": [Function],
        "onMonthChange": [Function],
        "onSelect": [Function],
        "onYearChange": [Function],
        "preventOpenOnFocus": false,
        "previousMonthButtonLabel": "Previous Month",
        "readOnly": false,
        "renderDayContents": [Function],
        "selected": undefined,
        "shouldCloseOnSelect": false,
        "showTimeSelect": false,
        "timeCaption": "Time",
        "timeIntervals": 30,
        "withPortal": false,
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`The FilterComponent component should not regress 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <FilterComponent
    filter={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <form
        autoComplete="off"
        className="filter-box"
        onChange={[Function]}
        onSubmit={[Function]}
      >
        <input
          className="search-input"
          name="searchTerm"
          type="text"
        />
        <div
          className="filter-button"
          data-toggle="filter-dropdown"
          onClick={[Function]}
        >
          <img
            alt="filter"
            className="filter-button-icon"
            data-toggle="filter-dropdown"
            src="https://img.icons8.com/small/100/000000/sorting-options.png"
          />
        </div>
      </form>,
      "className": "filter-container",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "autoComplete": "off",
        "children": Array [
          <input
            className="search-input"
            name="searchTerm"
            type="text"
          />,
          <div
            className="filter-button"
            data-toggle="filter-dropdown"
            onClick={[Function]}
          >
            <img
              alt="filter"
              className="filter-button-icon"
              data-toggle="filter-dropdown"
              src="https://img.icons8.com/small/100/000000/sorting-options.png"
            />
          </div>,
          false,
        ],
        "className": "filter-box",
        "onChange": [Function],
        "onSubmit": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "search-input",
            "name": "searchTerm",
            "type": "text",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <img
              alt="filter"
              className="filter-button-icon"
              data-toggle="filter-dropdown"
              src="https://img.icons8.com/small/100/000000/sorting-options.png"
            />,
            "className": "filter-button",
            "data-toggle": "filter-dropdown",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "filter",
              "className": "filter-button-icon",
              "data-toggle": "filter-dropdown",
              "src": "https://img.icons8.com/small/100/000000/sorting-options.png",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          "type": "div",
        },
        false,
      ],
      "type": "form",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <form
          autoComplete="off"
          className="filter-box"
          onChange={[Function]}
          onSubmit={[Function]}
        >
          <input
            className="search-input"
            name="searchTerm"
            type="text"
          />
          <div
            className="filter-button"
            data-toggle="filter-dropdown"
            onClick={[Function]}
          >
            <img
              alt="filter"
              className="filter-button-icon"
              data-toggle="filter-dropdown"
              src="https://img.icons8.com/small/100/000000/sorting-options.png"
            />
          </div>
        </form>,
        "className": "filter-container",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "autoComplete": "off",
          "children": Array [
            <input
              className="search-input"
              name="searchTerm"
              type="text"
            />,
            <div
              className="filter-button"
              data-toggle="filter-dropdown"
              onClick={[Function]}
            >
              <img
                alt="filter"
                className="filter-button-icon"
                data-toggle="filter-dropdown"
                src="https://img.icons8.com/small/100/000000/sorting-options.png"
              />
            </div>,
            false,
          ],
          "className": "filter-box",
          "onChange": [Function],
          "onSubmit": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "search-input",
              "name": "searchTerm",
              "type": "text",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <img
                alt="filter"
                className="filter-button-icon"
                data-toggle="filter-dropdown"
                src="https://img.icons8.com/small/100/000000/sorting-options.png"
              />,
              "className": "filter-button",
              "data-toggle": "filter-dropdown",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "filter",
                "className": "filter-button-icon",
                "data-toggle": "filter-dropdown",
                "src": "https://img.icons8.com/small/100/000000/sorting-options.png",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            "type": "div",
          },
          false,
        ],
        "type": "form",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;

exports[`The FilterDropdown component should not regress 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <FilterDropdown
    filters={
      Object {
        "automation-type": "",
        "date": Object {
          "from": "",
          "to": "",
        },
        "emailAutomation": "",
        "nokoAutomation": "",
        "search-by": "",
        "searchTerm": "",
        "showFilterDropdown": false,
        "slackAutomation": "",
      }
    }
    handleInputChange={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <FilterSection
          inputs={
            Array [
              Object {
                "checkCondition": false,
                "id": "search-by-fellow",
                "label": "fellow",
                "name": "search-by",
                "noLabel": false,
                "value": "fellow",
              },
              Object {
                "checkCondition": false,
                "id": "search-by-partner",
                "label": "partner",
                "name": "search-by",
                "noLabel": false,
                "value": "partner",
              },
            ]
          }
          section="search-by"
        />,
        <FilterSection
          inputs={
            Array [
              Object {
                "checkCondition": false,
                "id": "automation-type-onboarding",
                "label": "onboarding",
                "name": "automation-type",
                "noLabel": false,
                "value": "onboarding",
              },
              Object {
                "checkCondition": false,
                "id": "automation-type-offboarding",
                "label": "offboarding",
                "name": "automation-type",
                "noLabel": false,
                "value": "offboarding",
              },
            ]
          }
          section="automation-type"
        />,
        <AutomationStatus
          statusInputs={
            Array [
              Object {
                "checkboxes": Array [
                  Object {
                    "checkCondition": false,
                    "id": "emailAutomation-success",
                    "label": "",
                    "name": "emailAutomation",
                    "noLabel": true,
                    "value": "success",
                  },
                  Object {
                    "checkCondition": false,
                    "id": "emailAutomation-failure",
                    "label": undefined,
                    "name": "emailAutomation",
                    "noLabel": true,
                    "value": "failure",
                  },
                ],
                "section": "email",
              },
              Object {
                "checkboxes": Array [
                  Object {
                    "checkCondition": false,
                    "id": "slackAutomation-success",
                    "label": "",
                    "name": "slackAutomation",
                    "noLabel": true,
                    "value": "success",
                  },
                  Object {
                    "checkCondition": false,
                    "id": "slackAutomation-failure",
                    "label": undefined,
                    "name": "slackAutomation",
                    "noLabel": true,
                    "value": "failure",
                  },
                ],
                "section": "slack",
              },
              Object {
                "checkboxes": Array [
                  Object {
                    "checkCondition": false,
                    "id": "nokoAutomation-success",
                    "label": "",
                    "name": "nokoAutomation",
                    "noLabel": true,
                    "value": "success",
                  },
                  Object {
                    "checkCondition": false,
                    "id": "nokoAutomation-failure",
                    "label": undefined,
                    "name": "nokoAutomation",
                    "noLabel": true,
                    "value": "failure",
                  },
                ],
                "section": "noko",
              },
            ]
          }
        />,
        <AutomationDate
          dateInputs={
            Array [
              Object {
                "dateSelected": "",
                "onChange": [Function],
                "selection": "from",
              },
              Object {
                "dateSelected": "",
                "minDate": undefined,
                "onChange": [Function],
                "selection": "to",
              },
            ]
          }
        />,
        <button
          className="apply-filters"
          type="submit"
        >
          Apply Filters
        </button>,
      ],
      "className": "filter-dropdown-parent",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "inputs": Array [
            Object {
              "checkCondition": false,
              "id": "search-by-fellow",
              "label": "fellow",
              "name": "search-by",
              "noLabel": false,
              "value": "fellow",
            },
            Object {
              "checkCondition": false,
              "id": "search-by-partner",
              "label": "partner",
              "name": "search-by",
              "noLabel": false,
              "value": "partner",
            },
          ],
          "section": "search-by",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "inputs": Array [
            Object {
              "checkCondition": false,
              "id": "automation-type-onboarding",
              "label": "onboarding",
              "name": "automation-type",
              "noLabel": false,
              "value": "onboarding",
            },
            Object {
              "checkCondition": false,
              "id": "automation-type-offboarding",
              "label": "offboarding",
              "name": "automation-type",
              "noLabel": false,
              "value": "offboarding",
            },
          ],
          "section": "automation-type",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "statusInputs": Array [
            Object {
              "checkboxes": Array [
                Object {
                  "checkCondition": false,
                  "id": "emailAutomation-success",
                  "label": "",
                  "name": "emailAutomation",
                  "noLabel": true,
                  "value": "success",
                },
                Object {
                  "checkCondition": false,
                  "id": "emailAutomation-failure",
                  "label": undefined,
                  "name": "emailAutomation",
                  "noLabel": true,
                  "value": "failure",
                },
              ],
              "section": "email",
            },
            Object {
              "checkboxes": Array [
                Object {
                  "checkCondition": false,
                  "id": "slackAutomation-success",
                  "label": "",
                  "name": "slackAutomation",
                  "noLabel": true,
                  "value": "success",
                },
                Object {
                  "checkCondition": false,
                  "id": "slackAutomation-failure",
                  "label": undefined,
                  "name": "slackAutomation",
                  "noLabel": true,
                  "value": "failure",
                },
              ],
              "section": "slack",
            },
            Object {
              "checkboxes": Array [
                Object {
                  "checkCondition": false,
                  "id": "nokoAutomation-success",
                  "label": "",
                  "name": "nokoAutomation",
                  "noLabel": true,
                  "value": "success",
                },
                Object {
                  "checkCondition": false,
                  "id": "nokoAutomation-failure",
                  "label": undefined,
                  "name": "nokoAutomation",
                  "noLabel": true,
                  "value": "failure",
                },
              ],
              "section": "noko",
            },
          ],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "dateInputs": Array [
            Object {
              "dateSelected": "",
              "onChange": [Function],
              "selection": "from",
            },
            Object {
              "dateSelected": "",
              "minDate": undefined,
              "onChange": [Function],
              "selection": "to",
            },
          ],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Apply Filters",
          "className": "apply-filters",
          "type": "submit",
        },
        "ref": null,
        "rendered": "Apply Filters",
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <FilterSection
            inputs={
              Array [
                Object {
                  "checkCondition": false,
                  "id": "search-by-fellow",
                  "label": "fellow",
                  "name": "search-by",
                  "noLabel": false,
                  "value": "fellow",
                },
                Object {
                  "checkCondition": false,
                  "id": "search-by-partner",
                  "label": "partner",
                  "name": "search-by",
                  "noLabel": false,
                  "value": "partner",
                },
              ]
            }
            section="search-by"
          />,
          <FilterSection
            inputs={
              Array [
                Object {
                  "checkCondition": false,
                  "id": "automation-type-onboarding",
                  "label": "onboarding",
                  "name": "automation-type",
                  "noLabel": false,
                  "value": "onboarding",
                },
                Object {
                  "checkCondition": false,
                  "id": "automation-type-offboarding",
                  "label": "offboarding",
                  "name": "automation-type",
                  "noLabel": false,
                  "value": "offboarding",
                },
              ]
            }
            section="automation-type"
          />,
          <AutomationStatus
            statusInputs={
              Array [
                Object {
                  "checkboxes": Array [
                    Object {
                      "checkCondition": false,
                      "id": "emailAutomation-success",
                      "label": "",
                      "name": "emailAutomation",
                      "noLabel": true,
                      "value": "success",
                    },
                    Object {
                      "checkCondition": false,
                      "id": "emailAutomation-failure",
                      "label": undefined,
                      "name": "emailAutomation",
                      "noLabel": true,
                      "value": "failure",
                    },
                  ],
                  "section": "email",
                },
                Object {
                  "checkboxes": Array [
                    Object {
                      "checkCondition": false,
                      "id": "slackAutomation-success",
                      "label": "",
                      "name": "slackAutomation",
                      "noLabel": true,
                      "value": "success",
                    },
                    Object {
                      "checkCondition": false,
                      "id": "slackAutomation-failure",
                      "label": undefined,
                      "name": "slackAutomation",
                      "noLabel": true,
                      "value": "failure",
                    },
                  ],
                  "section": "slack",
                },
                Object {
                  "checkboxes": Array [
                    Object {
                      "checkCondition": false,
                      "id": "nokoAutomation-success",
                      "label": "",
                      "name": "nokoAutomation",
                      "noLabel": true,
                      "value": "success",
                    },
                    Object {
                      "checkCondition": false,
                      "id": "nokoAutomation-failure",
                      "label": undefined,
                      "name": "nokoAutomation",
                      "noLabel": true,
                      "value": "failure",
                    },
                  ],
                  "section": "noko",
                },
              ]
            }
          />,
          <AutomationDate
            dateInputs={
              Array [
                Object {
                  "dateSelected": "",
                  "onChange": [Function],
                  "selection": "from",
                },
                Object {
                  "dateSelected": "",
                  "minDate": undefined,
                  "onChange": [Function],
                  "selection": "to",
                },
              ]
            }
          />,
          <button
            className="apply-filters"
            type="submit"
          >
            Apply Filters
          </button>,
        ],
        "className": "filter-dropdown-parent",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "inputs": Array [
              Object {
                "checkCondition": false,
                "id": "search-by-fellow",
                "label": "fellow",
                "name": "search-by",
                "noLabel": false,
                "value": "fellow",
              },
              Object {
                "checkCondition": false,
                "id": "search-by-partner",
                "label": "partner",
                "name": "search-by",
                "noLabel": false,
                "value": "partner",
              },
            ],
            "section": "search-by",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "inputs": Array [
              Object {
                "checkCondition": false,
                "id": "automation-type-onboarding",
                "label": "onboarding",
                "name": "automation-type",
                "noLabel": false,
                "value": "onboarding",
              },
              Object {
                "checkCondition": false,
                "id": "automation-type-offboarding",
                "label": "offboarding",
                "name": "automation-type",
                "noLabel": false,
                "value": "offboarding",
              },
            ],
            "section": "automation-type",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "statusInputs": Array [
              Object {
                "checkboxes": Array [
                  Object {
                    "checkCondition": false,
                    "id": "emailAutomation-success",
                    "label": "",
                    "name": "emailAutomation",
                    "noLabel": true,
                    "value": "success",
                  },
                  Object {
                    "checkCondition": false,
                    "id": "emailAutomation-failure",
                    "label": undefined,
                    "name": "emailAutomation",
                    "noLabel": true,
                    "value": "failure",
                  },
                ],
                "section": "email",
              },
              Object {
                "checkboxes": Array [
                  Object {
                    "checkCondition": false,
                    "id": "slackAutomation-success",
                    "label": "",
                    "name": "slackAutomation",
                    "noLabel": true,
                    "value": "success",
                  },
                  Object {
                    "checkCondition": false,
                    "id": "slackAutomation-failure",
                    "label": undefined,
                    "name": "slackAutomation",
                    "noLabel": true,
                    "value": "failure",
                  },
                ],
                "section": "slack",
              },
              Object {
                "checkboxes": Array [
                  Object {
                    "checkCondition": false,
                    "id": "nokoAutomation-success",
                    "label": "",
                    "name": "nokoAutomation",
                    "noLabel": true,
                    "value": "success",
                  },
                  Object {
                    "checkCondition": false,
                    "id": "nokoAutomation-failure",
                    "label": undefined,
                    "name": "nokoAutomation",
                    "noLabel": true,
                    "value": "failure",
                  },
                ],
                "section": "noko",
              },
            ],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "dateInputs": Array [
              Object {
                "dateSelected": "",
                "onChange": [Function],
                "selection": "from",
              },
              Object {
                "dateSelected": "",
                "minDate": undefined,
                "onChange": [Function],
                "selection": "to",
              },
            ],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Apply Filters",
            "className": "apply-filters",
            "type": "submit",
          },
          "ref": null,
          "rendered": "Apply Filters",
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
